paimon-core/src/main/java/org/apache/paimon/io/KeyValueDataFileWriter.java

```diff
@@ -80,7 +82,8 @@ public KeyValueDataFileWriter(
             long schemaId,
             int level,
             String compression,
-            CoreOptions options) {
+            CoreOptions options,
+            boolean isCompact) {
```

Comment:
Maybe just pass `FileSource`? For all places in this PR.

Reply:
Fixed